### DISCUSSION TITLE
Deploy application from local machine with Dockerfile and custom context path

### DISCUSSION THREAD

**Discussion by melokki** [coollabsio/coolify#842] [Q&A]
Hello,
I just discovered coolify and I wanted to use it in my local development, but I am stuck on something
I wanted to deploy a new application using a `Dockerfile`, but as it seems, this feature is not what I thought it is. I copy/pasted my `Dockerfile` in the input and hit deploy, the build instantly failed because coolify does not know my local context to build the application. In my `Dockerfile` I am coping some file from local machine to the container. (won't do this in production, it's just for development purposes)

Is there a way to specify the context for a `Dockerfile` or simply upload a `Dockerfile`  and that path to be the build context??

Or should I look somewhere else for such solution?
Any ideas?

**Comment by peaklabs-dev** [comment]
@melokki Yes, this is possible in v4. Deploy via GitHub and select Dockerfile as the build method. Let me know if you need more help (use v4).
